core/src/main/java/org/apache/calcite/rel/rules/ReduceExpressionsRule.java

```diff
@@ -1069,6 +1139,117 @@ private void reduceCasts(RexCall outerCast) {
       }
     }
   }
+
+  /**
+   * An immutable object capturing the set of options to be used for reducing
+   * expressions.
+   */
+  public static final class ReductionOptions {
+    private final boolean matchNullability;
+    private final boolean unknownAsFalse;
+    private final boolean treatDynamicCallsAsNonConstant;
+
+    private ReductionOptions(
+        boolean matchNullability,
+        boolean unknownAsFalse,
+        boolean treatDynamicCallsAsNonConstant) {
```

Comment:
Have you considered adding a enum?
Frankly speaking, method calls like `(true, true, false)` are hard to read, and they are prone to errors.